### PR TITLE
Updates to the PowerShell modules

### LINKED ISSUE #4411: Add aliases to the cmdlets in Microsoft.Winget.Client PowerShell module
### Description of the new feature / enhancement

Add some default aliases for commands to reduce typing mistakes and aid use of the module on the cmdline

 - [ ] Add-WinGetSource
 - [ ] Assert-WinGetPackageManager
 - [ ] Disable-WinGetSetting
 - [ ] Enable-WinGetSetting
 - [ ] Export-WinGetPackage
 - [ ] Find-WinGetPackage
 - [ ] Get-WinGetPackage
 - [ ] Get-WinGetSettings
 - [ ] Get-WinGetSource
 - [ ] Get-WinGetUserSettings
 - [ ] Get-WinGetVersion
 - [ ] Install-WinGetPackage
 - [ ] Remove-WinGetSource
 - [ ] Repair-WinGetPackageManager
 - [ ] Reset-WinGetSource
 - [ ] Set-WinGetUserSettings
 - [ ] Test-WinGetUserSettings
 - [ ] Uninstall-WinGetPackage
 - [ ] Update-WinGetPackage

### Proposed technical implementation details

_No response_

### LINKED ISSUE #4680: Make Microsoft.WinGet.Client PowerShell cmdlets "singular" in name
### Description of the new feature / enhancement

The following cmdlets are plural in name and should be singular:
* Get-WinGetSettings
* Get-WinGetUserSettings
* Set-WinGetUserSettings
* Test-WinGetUserSettings

### Proposed technical implementation details

_No response_

### PR DESCRIPTION
**Pull request by kilasuit**
- [x] I have signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs).
- [x] This pull request is related to an issue
       - fixes #4411
       - fixes #4680  

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/4716)

---
edit by @denelon - formatting for checkboxes :)

### CODE REVIEW

**Comment by kilasuit** [comment]
FYI I could not build this small part locally at all getting this error no matter what I do to try and build it 
```
error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. 
Confirm that the expression in the Import declaration "\Microsoft.Cpp.Default.props" is correct, and that the file exists on disk.
```
I also was thinking that on push to this repo that a build would be automatically triggered, though looking at other pr's it's manually being kicked off using the `/azp run` mechanism (which I don't think I would be able to run)

Other than that this is ready to merge if a build completes no issue

**Comment by kilasuit** [comment]
was worth a shot 🤷🏻‍♂️

File: .github/actions/spelling/expect.txt
Comment (Trenly) [inline .github/actions/spelling/expect.txt:1]: Instead of adding the individual values to expect, it would be better to use `patterns.txt` since all the Aliases follow the same format of `\[Alias\([a-z]+\)\]`

Comment (kilasuit) [reply .github/actions/spelling/expect.txt:1]: I can add this in the expect file however this would then flag in the psd1 files for both these modules 
I will add it to patterns.txt, but am keeping the addition in the expects file too

**Comment by kilasuit** [comment]
Just working on adding the additional bits needed for #4680 to get this all added in a single PR

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs:16]: Remove the `Settings` constant.

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetUserSettingCmdlet.cs
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetUserSettingCmdlet.cs:17]: I don't see `Constants.cs` in the changes, which would be required to change the name and value here.

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs:17]: We should have the old name as an alias for all of these that are changing, no?  Or were these never available in a release version of the module?

File: src/PowerShell/Microsoft.WinGet.Client/ModuleFiles/Microsoft.WinGet.Client.psd1
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.Client/ModuleFiles/Microsoft.WinGet.Client.psd1:103]: Should this include the old names for the singularized cmdlets?

Comment (kilasuit) [reply src/PowerShell/Microsoft.WinGet.Client/ModuleFiles/Microsoft.WinGet.Client.psd1:103]: Yes it should, will include in the next update

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs
Comment (kilasuit) [reply src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs:17]: I was in the middle of fixing this last week & will be included in my next commit

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetUserSettingCmdlet.cs
Comment (kilasuit) [reply src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetUserSettingCmdlet.cs:17]: Thanks, will have a look over this and have that updated too.

**Comment by kilasuit** [comment]
@JohnMcPMS thanks for the comments, I'll get back to these during this week (hopefully tonight) as I have some updates on another device that was in progress at the end of the week.

**Comment by denelon** [comment]
@kilasuit thanks for jumping in here!

I think we'd want to make sure and update the appropriate "Help" files with the changes as well. Assuming this is all "non-breaking" changes, that can be a separate PR. I know we're also adding a new cmdlet for the WinGet Repair command.

**Comment by kilasuit** [comment]
So this should be good for merging now & thanks for the ping @denelon 

Only other thoughts 
- [ ] Is the file name changes for the depluralised cmdlet files ok or does this need reverting
- [ ] Is the change to the class name as per https://github.com/microsoft/winget-cli/pull/4716/commits/696bab0e0f0fec615bb49c9bf5c9b32c11b5281e ok?
- [ ] Do you want this rebased with cleaner commit history at all or is it fine as is?

**Comment by denelon** [comment]
If the cmdlets have been renamed to singular, I'd say we should keep them plural unless the help files get included in the PR. Probably best to do the rename in a separate PR so the help files don't break or confuse users. Get-Help and Get-Command should stay in sync.

I'll let one of our engineers make the call on the remaining "other thoughts".

**Comment by kilasuit** [comment]
Thought it best to add all in together so I have updated the helpfiles in this PR - though I can easily remove those commits if desired

**Comment by denelon** [comment]
OK, we're cutting a build with the current cmdlets and help. As soon as that is out, we can dig back through this PR to get updates for all the changes. I really appreciate the help here!

**Comment by JohnMcPMS** [comment]
Looking good to me; we need to fix some infrastructure issues and then we can run the build/test pass.

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/RepairWinGetPackageManagerCmdlet.cs
Comment (mdanish-kh) [inline src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/RepairWinGetPackageManagerCmdlet.cs:21]:
```suggestion
    [Alias("rpwgpm")]
```

Shouldn't this be without a `d`?

**Comment by denelon** [comment]
The following PR was just merged:
* #4781

Those examples should be included here.

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs:17]:
```
src\PowerShell\Microsoft.WinGet.Client.Cmdlets\Cmdlets\GetSettingCmdlet.cs(17,19): Error SA1001: Commas should be followed by whitespace. 
src\PowerShell\Microsoft.WinGet.Client.Cmdlets\Cmdlets\SetUserSettingCmdlet.cs(19,19): Error SA1001: Commas should be followed by whitespace. 
src\PowerShell\Microsoft.WinGet.Client.Cmdlets\Cmdlets\TestUserSettingCmdlet.cs(18,19): Error SA1001: Commas should be followed by whitespace.
```

**Comment by kilasuit** [comment]
> The following PR was just merged:
> 
>     * [Updates to PowerShell help docs #4781](https://github.com/microsoft/winget-cli/pull/4781)
> 
> 
> Those examples should be included here.

I just rebased from the master branch & resolved the last comment so all good to go :-)

File: src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/GetSettingCmdlet.cs:17]:
```suggestion
    [Alias("gwgse", "Get-WinGetSettings")]
```
Could this alias duplication be interfering with the cmdlet export?  The tests are failing with:
```
##[error] [-] Get|Enable|Disable-WinGetSetting.Get-WinGetSetting 5ms (3ms|1ms)
##[error]  CommandNotFoundException: The term 'Get-WinGetSetting' is not recognized as a name of a cmdlet, function, script file, or executable program.
##[error]  Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
##[error]  at <ScriptBlock>, D:\a\1\s\src\PowerShell\tests\Microsoft.WinGet.Client.Tests.ps1:646
```
Which looks like the correct command to me.

File: src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1:77]: I removed these asserts in a change that has been merged recently, as well as other significant changes to this file that are not reflected here.  I'm not sure exactly what has happened, but we will have to work out the conflicts here before it can be merged.

**Review by JohnMcPMS — DISMISSED** [review]

File: src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1
Comment (JohnMcPMS) [reply src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1:77]: I will try to resolve it after I check on my own PR, but I'm not sure if I can update this PR directly.

Comment (JohnMcPMS) [reply src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1:77]: Nope, I'm too deep in the changes... I removed these in my own PR that has not been merged yet.

**Comment by JohnMcPMS** [comment]
Build was successful with my changes.  @kilasuit, please sign off on the PR to indicate that you are done with changes and ready to merge.

**Comment by kilasuit** [comment]
Yeah all good with me @JohnMcPMS

**Review by JohnMcPMS — APPROVED** [review]

**Pull request merged**
